src/lib.rs

```diff
@@ -239,14 +258,28 @@ fn setup_bitcoind(
         log::info!("Creating a new watchonly wallet on bitcoind.");
         bitcoind.create_watchonly_wallet(&config.main_descriptor)?;
         log::info!("Watchonly wallet created.");
-    } else {
+    } else if !is_remote {
         #[cfg(windows)]
         if !cfg!(test) && !wo_path.exists() {
             return Err(StartupError::NoWatchonlyInDatadir);
         }
     }
     log::info!("Loading our watchonly wallet on bitcoind.");
-    bitcoind.maybe_load_watchonly_wallet()?;
+    if is_remote {
+        if let Err(e) = bitcoind.maybe_load_watchonly_wallet() {
+            let wallets = bitcoind.list_wallets();
+            // if a wallet w/ legacy name exist
+            if wallets.contains(&wo_path_str) && !wallets.contains(&new_wo_path_str) {
+                log::info!("Loading legacy wallet.");
+                bitcoind.change_watchonly_wallet_path(wo_path_str);
```

Comment:
I think it would be safer to use `BitcoinD::new()` when changing the wallet name as it also performs connection checks.